packages/cursorless-engine/src/processTargets/TargetPipeline.ts

```diff
@@ -0,0 +1,276 @@
+import { ImplicitTargetDescriptor, Modifier, Range } from "@cursorless/common";
+import { uniqWith, zip } from "lodash";
+import {
+  PrimitiveTargetDescriptor,
+  RangeTargetDescriptor,
+  TargetDescriptor,
+} from "../typings/TargetDescriptor";
+import { ProcessedTargetsContext } from "../typings/Types";
+import { Target } from "../typings/target.types";
+import { MarkStageFactory } from "./MarkStageFactory";
+import { ModifierStageFactory } from "./ModifierStageFactory";
+import { MarkStage, ModifierStage } from "./PipelineStages.types";
+import ImplicitStage from "./marks/ImplicitStage";
+import { ContainingTokenIfUntypedEmptyStage } from "./modifiers/ConditionalModifierStages";
+import { PlainTarget, PositionTarget } from "./targets";
+
+export class TargetPipeline {
+  constructor(
+    private modifierStageFactory: ModifierStageFactory,
+    private markStageFactory: MarkStageFactory,
+    private context: ProcessedTargetsContext,
```

Comment:
Notice that putting these functions onto a class allows us to avoid passing the `context` around so much by just making it a member variable.  We'll eventually split up / remove the `context` altogether, but leaving that for later